Add Sidenav bar to the docs

yoyo-go opened the pull request:
Made a Sidenav bar with dropdown list, so users can find out what they want among a large number of relations more quickly.

fcbond commented:
Hi,

we tried opening it the index here, and could not see the navbar.   How did you test it?  We just opened the index.html in a browser, ...

But more importantly, you should not make changes to index.html directly, instead you should change `gwadoc/doc_en.py` then build the documentation.   Changes to the index should be to components inside `docs/templates`.

If you are having trouble with installing the [build] extra, please file an issue or email us.

Looking forward to hearing from you.

yoyo-go commented:
> Hi,
> 
> we tried opening it the index here, and could not see the navbar. How did you test it? We just opened the index.html in a browser, ...

Thanks for your feedback. I have opened the index.html page on my computer, it works for both Chrome and Firefox...Anyway, the sidenav should look like below:

Dropdown close:
![dropdownClose](https://user-images.githubusercontent.com/56577574/84595066-8fd35b00-ae88-11ea-8ac3-1ed35c513443.PNG)

Dropdown open:
![dropdownOpen](https://user-images.githubusercontent.com/56577574/84595076-9a8df000-ae88-11ea-9dd8-cd0d1aa2cf11.PNG)

> But more importantly, you should not make changes to index.html directly, instead you should change `gwadoc/doc_en.py` then build the documentation. Changes to the index should be to components inside `docs/templates`.

I know that if I want to make content changes, I should go to doc_en.py such as add content to the relation **Meronym**, and for the format, go to the template since it will generate the index automatically. But since I am not formally onboard for this project, I just want to let you have a look at the design of the sidnav first, but now you can not see it... Don't know why... TAT

> If you are having trouble with installing the [build] extra, please file an issue or email us.

I am trying to work out this in virtual environment, if I fail I will come to you. Thanks for that.

fcbond commented:
Our bad, we didn't switch to the correct branch!  Once we did, we can see the navbar. 

There are some aesthetic issues (maybe only show name, not the internal name so it fits, boxed plus and minus instead of down arrow), but these are relatively easy fixes.   I think having the menu where you can see it all the time is a definite enhancement.

However, I do miss having the compact summary of all relations at the top.   Do you think it makes sense to have it in two places?  They serve different functions, and I think don't get in each other's way.

Yours,

yoyo-go commented:
> There are some aesthetic issues (maybe only show name, not the internal name so it fits, boxed plus and minus instead of down arrow), but these are relatively easy fixes. I think having the menu where you can see it all the time is a definite enhancement.

Ok, feedback well received, I will adjust it. GSoD is starting applying now, we writers need to create a project proposal and submit it before July, so I will deal with my proposal first and go back to this later.

> However, I do miss having the compact summary of all relations at the top. Do you think it makes sense to have it in two places? They serve different functions, and I think don't get in each other's way.

Ok, We can move the Table of content back to the webpage, and add it as "Table of content" at the very top of the navbar. 

Besides, [build] is installed successfully, but when I try to run below command line, it throws an error:
![00](https://user-images.githubusercontent.com/56577574/84597483-ce701200-ae96-11ea-925e-54ef1aaecd5f.PNG)

Best regards,
Yoyo

fcbond commented:
HI,

I think you should run 'set PYTHONIOENCODING=utf-8', or change your locale
to a unicode locale.

The proposal is not due until July 9th (and does not get sent to us until
then) so there is some time left.


On Sun, Jun 14, 2020 at 11:40 PM yoyo-go <notifications@github.com> wrote:

> There are some aesthetic issues (maybe only show name, not the internal
> name so it fits, boxed plus and minus instead of down arrow), but these are
> relatively easy fixes. I think having the menu where you can see it all the
> time is a definite enhancement.
>
> Ok, feedback well received, I will adjust it. GSoD is starting applying
> now, we writers need to create a project proposal and submit it before
> July, so I will deal with my proposal first and go back to this later.
>
> However, I do miss having the compact summary of all relations at the top.
> Do you think it makes sense to have it in two places? They serve different
> functions, and I think don't get in each other's way.
>
> Ok, We can move the Table of content back to the webpage, and add it as
> "Table of content" at the very top of the navbar.
>
> Besides, [build] is installed successfully, but when I try to run below
> command line, it throws an error:
> [image: 00]
> <https://user-images.githubusercontent.com/56577574/84597483-ce701200-ae96-11ea-925e-54ef1aaecd5f.PNG>
>
> Best regards,
> Yoyo
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/globalwordnet/gwadoc/pull/17#issuecomment-643783615>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIPZRRUSFV3NNUWD75DPFTRWTVOBANCNFSM4NWDT2KA>
> .
>


-- 
Francis Bond <http://www3.ntu.edu.sg/home/fcbond/>
Division of Linguistics and Multilingual Studies
Nanyang Technological University

yoyo-go commented:
> HI, I think you should run 'set PYTHONIOENCODING=utf-8', or change your locale to a unicode locale. 

Thanks for your advice, it now works! :）

Besides the fix of the webpage, have you reviewed my writing sample of **Meronym** and **Holonym**, I have sent it through email, also attached here as convenience. 
[Writing sample-0602.docx](https://github.com/globalwordnet/gwadoc/files/4781396/Writing.sample-0602.docx)

I have added EWN test 21 and example to these two relations, which is cited from the EWN general document. I also rewritten all the EWN tests, don't know whether it is proper for the docs, or just directly copy them from EWN without changing a word?

If the content is approved by you, I can add them to the `doc_en.py` this time.

yoyo-go commented:
> There are some aesthetic issues (maybe only show name, not the internal name so it fits, boxed plus and minus instead of down arrow), but these are relatively easy fixes. I think having the menu where you can see it all the time is a definite enhancement.
> 
> However, I do miss having the compact summary of all relations at the top. Do you think it makes sense to have it in two places? They serve different functions, and I think don't get in each other's way.
> 
Hi, I have made a new commit to this branch to revise the sidebar per your feedback, hope you can see it. :)
